src/frontend/components/UI/QuestsViewer/components/QuestDetailsWrapper/index.tsx

```diff
@@ -25,23 +31,82 @@ import { getRewardCategory } from 'frontend/helpers/getRewardCategory'
 import { getDecimalNumberFromAmount } from '@hyperplay/utils'
 import { useFlags } from 'launchdarkly-react-client-sdk'
 import { getPlaystreakArgsFromQuestData } from 'frontend/helpers/getPlaystreakArgsFromQuestData'
+import { createPublicClient } from 'viem'
+import { chainMap, parseChainMetadataToViemChain } from '@hyperplay/chains'
+import { InfoAlertProps } from '@hyperplay/ui/dist/components/AlertCard'
 
 export interface QuestDetailsWrapperProps {
   selectedQuestId: number | null
 }
 
+const averageGasPerFunction: Record<string, number> = {
+  ERC1155: 102_470,
+  ERC721: 107_567,
+  ERC20: 98_507
+}
+
+async function getRewardClaimGasEstimation(reward: Reward) {
+  if (!reward.chain_id) {
+    throw Error(`chain_id is not set for reward: ${reward.id}`)
+  }
+
+  const chainMetadata = chainMap[reward.chain_id]
+
+  if (!chainMetadata) {
+    throw Error(`chainMetadata is not set for reward: ${reward.id}`)
+  }
+
+  const viemChain = parseChainMetadataToViemChain(chainMetadata)
+
+  const publicClient = createPublicClient({
+    // @ts-expect-error: chain types are valid
+    chain: viemChain,
+    transport: http()
+  })
+
+  let gasPerFunction
+
+  switch (reward.reward_type) {
+    case 'ERC1155':
+    case 'ERC721':
+    case 'ERC20':
+      gasPerFunction = averageGasPerFunction[reward.reward_type]
```

Comment:
what do you think about just increasing this by 50% to be on the safe side so when we make contract iterations in the future, this code will still probably work. 

this will ensure that the user has enough gas to increase the gas price a bit in the case of a gas price spike too